src/main/java/org/prebid/server/handler/NotificationEventHandler.java

```diff
@@ -26,107 +34,167 @@
         }
 
         final String bidId = queryParameters.get(BID_ID_PARAMETER);
         if (StringUtils.isBlank(bidId)) {
-            throw new InvalidRequestException("bidid is required query parameter and can't be empty.");
+            throw new InvalidRequestException("'bidid' is required query parameter and can't be empty");
         }
 
-        final String bidder = queryParameters.get(BIDDER_PARAMETER);
-        if (StringUtils.isBlank(bidder)) {
-            throw new InvalidRequestException("bidder is required query parameter and can't be empty.");
+        final String format = queryParameters.get(FORMAT_PARAMETER);
+        if (StringUtils.isNotBlank(format) && ObjectUtils.notEqual(format, BLANK_FORMAT)
+                && ObjectUtils.notEqual(format, PIXEL_FORMAT)) {
+            throw new InvalidRequestException(String.format("'format' is required query parameter. Possible values "
+                    + "are %s and %s, but was %s", BLANK_FORMAT, PIXEL_FORMAT, format));
+        }
+
+        final String analytics = queryParameters.get(ANALYTICS_PARAMETER);
+        if (StringUtils.isNotBlank(analytics) && ObjectUtils.notEqual(analytics, ENABLED_ANALYTICS)
+                && ObjectUtils.notEqual(analytics, DISABLED_ANALYTICS)) {
+            throw new InvalidRequestException(String.format("'analytics' is required query parameter. Possible values "
+                    + "are %s and %s, but was %s", ENABLED_ANALYTICS, DISABLED_ANALYTICS, analytics));
+        }
+    }
+
+    private static void validateParametersForUnauthorizedError(MultiMap queryParameters) {
+        final String account = queryParameters.get(ACCOUNT_PARAMETER);
+        if (StringUtils.isBlank(account)) {
+            throw new InvalidRequestException("'account' is required query parameter and can't be empty");
         }
-        return NotificationEvent.of(type, bidId, bidder);
     }
 
-    private void validateEventRequestQueryParams(String format) {
-        if (format != null && !trackingPixels.containsKey(format)) {
-            throw new InvalidRequestException(String.format(
-                    "'format' query parameter can have one of the next values: %s", trackingPixels.keySet()));
+    private Future<Boolean> isAccountEventEnabled(String accountId) {
+        return applicationSettings.getAccountById(accountId, timeoutFactory.create(DEFAULT_TIMEOUT))
+                .map(Account::getEventsEnabled)
+                .otherwise(throwable -> fallbackResult(accountId, throwable));
+    }
+
+    private void handleEvent(AsyncResult<Boolean> isEnabledResult, boolean isAnalyticsEnabled, String format,
+                             RoutingContext context) {
+        final HttpServerResponse response = context.response();
+        if (isEnabledResult.result()) {
+            if (isAnalyticsEnabled) {
+                analyticsReporter.processEvent(makeNotificationEvent(context));
+                respondWithOkStatus(response, format);
+            }
```

Comment:
What is the response for the case when analytics is disabled for account?

Reply:
401. 
I changed names to be more clearer what happening.

Reply:
Do we really need to fetch account data if query string has `x=0`?